src/python/zquantum/core/utils.py

```diff
@@ -387,6 +387,19 @@ def save_list(array, filename, artifact_name=""):
     with open(filename, "w") as f:
         f.write(json.dumps(dictionary, indent=2))
 
+def save_generic_dict(dictionary, filename):
+    """Save dictionary as json
+
+    Args:
+        dictionary (dict): the dict containing the data
+    """
+    dictionary_stored = {}
+    dictionary_stored['schema'] = SCHEMA_VERSION + '-dict'
+    dictionary_stored['dict'] = dictionary
```

Comment:
This seems a bit odd - why not just put `'schema'` as one of the keys in the original dict? Pinging @mstechly for input